### ISSUE TITLE
Add back some QN Index functions from V2

### ISSUE THREAD

**Issue by mtfishman** [ITensor/ITensor#230]
Some QN Index functions were removed in the V2 -> V3 transition, such as `offset()` and `findByQN()`, which would be good to add back in some form for V3.

**Comment by emstoudenmire** [comment]
Agreed, although I think we could add these as well for version 3.0.1

**Comment by mtfishman** [comment]
Yeah, just putting it here as a reminder. I added the `V3` label just to help specify which issues are specific to V3 (as opposed to V2), not as a marker that it is necessary for the V3 release.
